bentoml/yatai/client/bento_repository_api.py

```diff
@@ -344,62 +346,79 @@ def list(
+            bento_name: string
+            bento_version: string
+            prune: boolean, Set True to delete all BentoService
+            require_confirm: boolean
         Example:
         >>>
-        >>> yatai_client = get_yatai_client()
-        >>> yatai_client.repository.delete('my_service:version')
-        """
-        track('py-api-delete')
-        if ':' not in bento:
-            raise BentoMLException(
-                'BentoService name or version is missing. Please provide in the '
-                'format of name:version'
-            )
-        name, version = bento.split(':')
-        result = self.yatai_service.DangerouslyDeleteBento(
-            DangerouslyDeleteBentoRequest(bento_name=name, bento_version=version)
-        )
-        if result.status.status_code != yatai_proto.status_pb2.Status.OK:
-            error_code, error_message = status_pb_to_error_code_and_message(
-                result.status
-            )
-            raise BentoMLException(
-                f'Failed to delete Bento {bento} {error_code}:{error_message}'
-            )
-
-    def prune(self, bento_name=None, labels=None):
-        """
-        Delete all BentoServices that matches the specified criteria
-
-        Args:
-            bento_name: optional
-            labels: optional
-
-        Example:
-
         >>> yatai_client = get_yatai_client()
         >>> # Delete all bento services
-        >>> yatai_client.repository.prune()
-        >>> # Delete bento services that matches with the label `ci=failed`
-        >>> yatai_client.repository.prune(labels='ci=failed')
+        >>> yatai_client.repository.delete(prune=True)
+        >>> # Delete bento service with name is `IrisClassifier` and version `0.1.0`
+        >>> yatai_client.repository.delete(
+        >>>     bento_name='IrisClassifier', bento_version='0.1.0'
+        >>> )
+        >>> # Delete all bento services with name 'MyService`
+        >>> yatai_client.repository.delete(bento_name='MyService')
+        >>> # Delete all bento services with labels match `ci=failed` and `cohort=20`
+        >>> yatai_client.repository.delete(labels='ci=failed, cohort=20')
         """
-        track('py-api-prune')
-        list_bentos_result = self.list(bento_name=bento_name, labels=labels,)
-        if list_bentos_result.status.status_code != yatai_proto.status_pb2.Status.OK:
-            error_code, error_message = status_pb_to_error_code_and_message(
-                list_bentos_result.status
+        track('py-api-delete')
+        bento_delete_list = []
+        if prune is True:
+            bento_delete_list = self.list()
+            logger.info(f'Deleting all {len(bento_delete_list)} BentoML saved bundles.')
+        else:
+            if bento_tag:
+                bento_delete_list.append(self.get(bento_tag))
+                logger.info(f'Deleting saved Bento bundle {bento_tag}')
+            if bento_name is not None and bento_version is not None:
```

Comment:
use `elif` here.  bento_tag and name/version can't be applied at once. otherwise it's very confusing.